### PR TITLE
Improve sending emails for EmailSendMixin

### LINKED ISSUE #2426: Errors in forms on test server
When filling out https://test-amy2.carpentries.org/forms/request_training/ 
we get the error:

```
AnymailRequestsAPIError at /forms/request_training/
Mailgun API response 401 (Unauthorized): 'Forbidden'
```

While the email should not go out, we should still be able to create forms in the test environment.

### PR DESCRIPTION
**Pull request by pbanaszkiewicz**
Silence the email errors in views using the EmailSendMixin.

This fixes #2426.

### CODE REVIEW

**Comment by pbanaszkiewicz** [comment]
@elichad ready for your review.

File: amy/extforms/views.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ class WorkshopRequestCreate(
     page_title = "Request a Carpentries Workshop"
     template_name = "forms/workshoprequest.html"
     success_url = reverse_lazy("workshop_request_confirm")
-    email_fail_silently = False
```
Comment (elichad) [inline amy/extforms/views.py:82]: Is there a way to do this only in non-production environments?

Comment (pbanaszkiewicz) [reply amy/extforms/views.py:82]: We can make condition based on the environment, just like in the case of top banner in test-amy or local env, but I'm not sure this is a good idea.

If the fail is silenced in test-amy then the forms should work (Maneesha's issue is fixed). In production we never had any issues with Mailgun, so this was never a problem.

**Comment by elichad** [comment]
Pinging @maneesha / @froggleston - this PR fixes a problem where we can't submit workshop request forms on test-amy, but it does so by ignoring errors from sending these emails in general.
We haven't encountered these errors in production, but they could appear under rare circumstances (e.g. if the Mailgun server goes down).

What are your thoughts on taking this approach until the new email system is ready?

**Comment by maneesha** [comment]
I am not comfortable with silent failures so I would like to see this ignored in test and local builds only.  Errors should be raised in production.  @pbanaszkiewicz  can you share why it is not a good idea to make it conditional based on environment?

**Comment by pbanaszkiewicz** [comment]
@elichad @maneesha @froggleston I'm going to decline this PR. I changed a single deployment environment variable (https://github.com/carpentries/systems/commit/12539f7647e81d8a73e3159f05e4a0a69c18fbc7) that controls the system used for sending emails in test-amy2. The reason is that:

1. I don't want to keep conditions in code that change the way it works based on the environment (I want to keep staging and production parity)
2. We're not sending real emails in that environment.

**Pull request closed without merging**